src/Build/BuildCheck/Infrastructure/BuildCheckBuildEventHandler.cs

```diff
@@ -45,12 +50,38 @@ internal class BuildCheckBuildEventHandler
 
     public void HandleBuildEvent(BuildEventArgs e)
     {
+        if (
```

Comment:
nit: consider adding a comment before this code for its intent?
```suggestion
        // Skip BuildCheck on restore
        if (
```